### PR TITLE
Remove an aggressive assert

### PR DESCRIPTION
**Pull request by kapilash**
Fix for #6199 

cc @stephentoub @vijaykota

### CODE REVIEW

File: src/Native/System.Net.Security.Native/pal_gssapi.cpp [hunk 1]
```diff
@@ -39,23 +39,14 @@ static char gss_mech_value[] = "\x2b\x06\x01\x05\x05\x02"; // Binary representat
...
-        targetBuffer->length = 0;
-        targetBuffer->data = nullptr;
-    }
-    else
-    {
-        assert(gssBuffer->value != nullptr || gssBuffer->length == 0);
-        targetBuffer->length = gssBuffer->length;
-        targetBuffer->data = static_cast<uint8_t*>(gssBuffer->value);
-    }
+    assert (gssBuffer->value == nullptr || gssBuffer->length > 0);
```
Comment (stephentoub) [inline src/Native/System.Net.Security.Native/pal_gssapi.cpp]: Nit: unnecessary space before the paren

**Comment by stephentoub** [comment]
LGTM

File: src/Native/System.Net.Security.Native/pal_gssapi.cpp (hunk 1 above)
Comment (kapilash) [inline src/Native/System.Net.Security.Native/pal_gssapi.cpp]: sorry, fixed it and squashed.

**Comment by vijaykota** [comment]
LGTM Thanks for the quick fix @kapilash

**Pull request merged**